File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        if (k <= 1) return 0;
        int n=nums.length;
        int i=0;
        int j=0;
        int mult=1;
        int result=0;

        while(j<n){
            mult*=nums[j];

            while(mult>=k){
                mult/=nums[i];
                i++;
            }
            result+=(j-i+1);
            j++;
        }
        return result;
    }
}

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        if(k<=1):
            return 0
        n=len(nums)
        i,j,result,mult=0,0,0,1
        while(j<n):
            mult*=nums[j]

            while(mult>=k):
                mult/=nums[i]
                i+=1
            result+=(j-i+1)
            j+=1
        return result